driver-core/src/main/com/mongodb/operation/FsyncUnlockOperation.java

```diff
@@ -35,9 +35,9 @@
  * @mongodb.driver.manual reference/command/fsyncUnlock/ fsyncUnlock command
  * @since 3.2
  */
-public class FsyncUnlockOperation implements WriteOperation<BsonDocument> {
+public class FsyncUnlockOperation implements ReadOperation<BsonDocument> {
```

Comment:
It's ugly, but to avoid breaking binary compatibility this class should implement both ReadOperation and WriteOperation.   WriteOperation implementation can be removed in the next major release.